packages/nimble-components/src/select/index.ts

```diff
@@ -969,39 +1009,92 @@ export class Select
         this.updateListboxMaxHeightCssVariable();
     }
 
+    private isOptionHidden(option: ListOption): boolean {
```

Comment:
nit: given that we have so many concepts of "hidden" now (option attribute, option filtered out, group attribute) it would be helpful to give this a more precise name. `shouldBeVisuallyHidden()`? `isOptionHiddenOrFilteredOut()`?